data-prepper-plugins/log-generator-source/src/test/java/com/amazon/dataprepper/plugins/source/loggenerator/LogGeneratorSourceTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source.loggenerator;
+
+
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.configuration.PluginModel;
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.plugin.PluginFactory;
+import com.amazon.dataprepper.model.record.Record;
+import com.amazon.dataprepper.plugins.buffer.blockingbuffer.BlockingBuffer;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.concurrent.TimeoutException;
+import java.time.Duration;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.lenient;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.Assert.assertEquals;
+
+public class LogGeneratorSourceTest {
+    private LogGeneratorSourceConfig sourceConfig;
+    private PluginMetrics pluginMetrics;
+    private PluginFactory pluginFactory;
+    private LogGeneratorSource logGeneratorSource;
+
+    @BeforeEach
+    public void setup() {
+        sourceConfig = mock(LogGeneratorSourceConfig.class);
+        pluginMetrics = mock(PluginMetrics.class);
+        pluginFactory = mock(PluginFactory.class);
```

Comment:
Nit: It's probably preferred to use inline mocks. You can add this dependency to your `build.gradle`: `testImplementation "org.mockito:mockito-inline:${versionMap.mockito}"`, annotate the class with `@ExtendWith(MockitoExtension.class)`, and then use the `@Mock` annotation on your private variables.